inspect_byproducts.py

```diff
@@ -0,0 +1,180 @@
+      Raises KeyError, in case the field corresponding to the key
+      in the dictionary is empty
+
+    <Returns>
+      Boolean
+    """
+
+    imported_link = link_import.read_from_file(link)
+
+    std_out_err = imported_link.byproducts[std]
+
+    if operator == 'is':
+      if std_out_err == inputstring:
+        return True
+
+    elif operator == 'is not':
+      if std_out_err != inputstring:
+        return True
+
+    elif operator == 'contains':
+      if std_out_err.find(inputstring) != -1:
+        return True
+
+    elif operator == 'contains not':
+      if std_out_err.find(inputstring) == -1:
+        return True
+    else:
+      raise Exception('Invalid operator')
+
+    return False
+
+def parse_args():
+    """
+    <Purpose>
+      A function which parses the user supplied arguments
+
+    <Arguments>
+      None
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Parsed arguments (args object)
+
+     """
+    parser = argparse.ArgumentParser(
+        description="Inspects the byproducts of a step")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "%(prog)s --link <path to link metadata>\n{0}"
+                    "[--stdout | --stderr]\n{0}"
+                    "[--is | --is-not | --contains | --contains-not]\n{0}"
+                    "string\n{0}"
+                    "[--verbose]\n\n"
+                    .format(lpad))
+
+    in_toto_args = parser.add_argument_group("in-toto-inspection options")
+
+    in_toto_args.add_argument("-l", "--link", type=str, required=True,
+                              help="Link metadata file to use for inspection "
+                              "of the step")
+
+    in_toto_args.add_argument("-st", "--outerr",
+                              type=str, required=True, help="when stdout or "
+                              "stderr is a byproduct")
+
+    in_toto_args.add_argument("-o", "--operator",
+                              type=str, required=True, help="whether the "
+                              "stdout or stderr is, is not,"
+                              "contains, contains not, the input string")
+
+    in_toto_args.add_argument("string", type=str,
+                              help="The string to which the return value "
+                              "should be compared")
+
+    in_toto_args.add_argument("-v", "--verbose", dest="verbose",
+                              help="Verbose execution.", default=False,
+                              action="store_true")
+
+    args = parser.parse_args()
+    args.operator = args.operator.lower()
+    args.outerr = args.outerr.lower()
+
+    return args
+
+def main():
+    """
+    First calls parse_args() to parse the arguments and then calls
+    inspect_byproducts to inspect the byproducts
+
+    """
+
+    args = parse_args()
+
+    if (args.outerr != 'stdout') & (args.outerr != 'stderr'):
+      print('Please specify only one of the following - stdout, stderr')
+      sys.exit(3)
```

Comment:
You are still using bitwise `&`, but you should use logical `&&`. [Argparser's choice argument](https://github.com/in-toto/in-toto-inspections/pull/2#discussion_r121991634) would eliminate the need for this check.